great_expectations/validator/validation_graph.py

```diff
@@ -28,20 +40,199 @@ def id(self):
 
 
 class ValidationGraph:
-    def __init__(self, edges: Optional[List[MetricEdge]] = None) -> None:
+    def __init__(
+        self,
+        edges: Optional[List[MetricEdge]] = None,
+        execution_engine: Optional[ExecutionEngine] = None,
+    ) -> None:
         if edges:
             self._edges = edges
         else:
             self._edges = []
 
         self._edge_ids = {edge.id for edge in self._edges}
 
+        self._execution_engine = execution_engine
+
+    @property
+    def edges(self):
+        return self._edges
+
+    @property
+    def edge_ids(self):
+        return {edge.id for edge in self._edges}
+
+    @property
+    def execution_engine(self) -> ExecutionEngine:
+        """Returns "ExecutionEngine" object, used for resolving "MetricConfiguration" objects in graph."""
+        return self._execution_engine
+
+    @execution_engine.setter
+    def execution_engine(self, value: ExecutionEngine):
+        self._execution_engine = value
```

Comment:
Why do we use `@propety` here?
Functionally this is the same as just having a `self.execution_engine` attribute that we access and modify directly.

Reply:
@Kilo59 Any idea why this module is not getting type checked automatically?  What should be added and where to force type checking on new modules that we create?  I am looking into your other questions in this comment.  Thank you!

Reply:
@Kilo59 This property is not used, so I deleted the related code.  Thanks for the great catch!